ams/TemplateMapper.cs

```diff
@@ -124,11 +124,11 @@ public string ExpandTemplate(string template, Func<string, string?> valueExtract
                     case "AssetId":
                         return (asset.Data.AssetId ?? Guid.Empty).ToString();
                     case "AssetName":
-                        return asset.Data.Name;
+                        return asset.Data.Name.Substring(0, Math.Min(asset.Data.Name.Length, 63));
```

Comment:
I guess this applies to the usage of ${AssetName} as part of the container name (e.g. before the first '/' of the full path)

A couple of notes:

1.  using${AssetName} as part of the path after the container name doesn't require trimming, so not sure if we want to apply trimming uniformly.
2.  just trimming $AssetName to 64 only fixes the case where user uses ${AssetName} by itself for the path template, e.g. ${AssetName}/../otherpath.  It doesn't fix for example, my${AssetName}/../otherpath as this requires trimming to 62 characters.
3. using ${AssetName} with trimming as the container name might hit issues where two unique asset names become the same container name (e.g. two asset names with the same first 64 characters but differs afterwards).  This is probably not a desired result so one should really avoid using AssetName in the container part of the path.

So if we still want to trim the AssetName after considering 3, then maybe a better place for trimming is to do it after line 114 of 

    public (string Container, string Prefix) ExpandPathTemplate(string template, Func<string, string?> extractor)
    
and just trim the containerName variable (this fixes 1, 2)

Reply:
Suggestion: Instead of silently trimming shall we just give a warning if ${AssetsName} is used as part of the container name?  We can just run as is and let the migration fail (we should fail early instead of waiting for first upload)

Reply:
@duggaraju in VI's case we have assets with too long names, so we must be able to trim.
@pohhsu do you see any other option rather than risking with having a conflict?